evalml/pipelines/components/transformers/preprocessing/delayed_feature_transformer.py

```diff
@@ -85,7 +85,7 @@ def transform(self, X, y=None):
         if X is None:
             X = pd.DataFrame()
         # Normalize the data into pandas objects
-        X_ww = infer_feature_types(X)
+        X_ww = infer_feature_types(X).ww.copy()
```

Comment:
This is a mistake I made in my components pr. Before we use `assign` which creates a new dataframe. Since we're not doing that, we need to create the copy ourselves. This is important so that 

```python
pipeline.predict(X, y)
pipeline.predict(X, y)
```

works. Pipeline tests caught this but I added a unit test in the delayed_feature_transformer unit tests as well.

Reply:
Awesome, so this makes sure that we don't change the original X input. Did we need to do this in the other components as well?

Reply:
Great question @bchen1116 ! I am adding coverage for the datetime featurizer and the text featurizer.